crates/bevy_diagnostic/src/lib.rs

```diff
@@ -15,10 +16,48 @@ pub struct DiagnosticsPlugin;
 
 impl Plugin for DiagnosticsPlugin {
     fn build(&self, app: &mut App) {
-        app.init_resource::<Diagnostics>();
+        app.init_resource::<Diagnostics>()
+            .add_startup_system(log_system_info);
     }
 }
 
 /// The width which diagnostic names will be printed as
 /// Plugin names should not be longer than this value
 pub const MAX_DIAGNOSTIC_NAME_WIDTH: usize = 32;
+
+#[derive(Debug)]
+// This is required because the Debug trait doesn't detect it's used when it's only used in a print :(
+#[allow(dead_code)]
+struct SystemInfo {
+    os: String,
+    kernel: String,
+    cpu: String,
+    core_count: String,
+    memory: String,
+}
+
+fn log_system_info() {
+    use sysinfo::{CpuExt, SystemExt};
+
+    let mut sys = sysinfo::System::new();
+    sys.refresh_cpu();
+    sys.refresh_memory();
+
+    let info = SystemInfo {
+        os: sys
+            .long_os_version()
+            .unwrap_or_else(|| String::from("not available")),
+        kernel: sys
+            .kernel_version()
+            .unwrap_or_else(|| String::from("not available")),
+        cpu: sys.global_cpu_info().brand().trim().to_string(),
+        core_count: sys
+            .physical_core_count()
+            .map(|x| x.to_string())
+            .unwrap_or_else(|| String::from("not available")),
+        // Convert from Bytes to GibiBytes since it's probably what people expect most of the time
+        memory: format!("{:.1} GiB", sys.total_memory() as f64 * 9.31323e-10),
```

Comment:
Wouldn't `sys.total_memory() as f64 / 1024 / 1024 / 1024` be easier to understand?

Reply:
Yeah, good idea, much easier to read than the magic number, not sure why I didn't do that in the first place.